### PR TITLE
Hca add spacing to last input

### PR DESCRIPTION
**Pull request by akainic**
This might be challenging to review because I restructured a lot of the jsx in the components. Main changes: 
- Structural changes due to removing as many uses of Foundation classes (`.row`, `.columns`) as possible
- Added a new class (`.input-section`) to handle the spacing issue that U.S. Web Design Standards has of adding margin to the label instead of margin bottom to the input, resulting in no space between the last input in a section and the subsequent header or button. Each block of questions that is followed by either a header or a button should be wrapped in a `div` with this class. Will submit an issue to USWDS.
- Changed `DateInput` to accept a label prop and added a test to check that label is a string.

### CODE REVIEW

File: _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx [hunk 1]
```diff
@@ -18,141 +18,123 @@ class AnnualIncomeSection extends React.Component {
...
-              <p>
-                <strong>
-                You are not required to enter financial information because you
-                indicated you are receiving a VA pension.
-                </strong>
-              </p>
-            }
-          </div>
+        <h4>Annual Income</h4>
+        {this.props.external.receivesVaPension === true &&
```
Comment (awong-dev) [inline _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx]: We should move this out so the conditional span is created outside of the return() statement and put in a variable.  This sort of conditional logic with a short-circuit and works, but it's a bit hard to read to me as it's too many lines.

I would prefer pulling the static content out into a variable which just gets default initialized to null or an empty string.

File: _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx [hunk 2]
```diff
@@ -18,141 +18,123 @@ class AnnualIncomeSection extends React.Component {
...
+          <p>
+            <strong>
+            You are not required to enter financial information because you
+            indicated you are receiving a VA pension.
+            </strong>
+          </p>
+        }
```
Comment (awong-dev) [inline _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx]: Do we need to exclude all the later stuff if the above is true?

**Comment by awong-dev** [comment]
LGTM with 2 comments to address in follow up PR. Merging.

File: _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx (hunk 2 above)
Comment (akainic) [inline _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx]: Probably, but we have an UX problem if we do that: if for some the user mistakingly indicated they received a VA pension and now realized they want to change their answer, the only way for them to do that is to press the back button 10+ times until they get back to the personal information section where that question is asked. We're thinking we might want to find a way to link back to that question more easily or even include it here if they want to change their response, but that's a bigger task to fix. I'll add a TODO to address later.

**Pull request merged**